File: typescript/base-hmi-types/src/screens/widgets/HmiAlarmIndicator.ts
import { HmiColor, HmiProperty } from "../base.js";
import { HmiSimpleScreenItemBase } from "../base/HmiSimpleScreenItemBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiAlarmIndicator extends HmiSimpleScreenItemBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiAlarmIndicator;
  }

  isFlashingRequired?: HmiProperty<boolean>;
  flashingColor?: HmiProperty<HmiColor>;
  flashingRate?: HmiProperty<number>;
  alarmState?: HmiProperty<number>;
  noAlarmState?: HmiProperty<number>;
  numberOfAlarms?: HmiProperty<number>;
  showAcknowledgedAlarmClasses?: HmiProperty<number[]>;
  showPendingAlarmClasses?: HmiProperty<number[]>;
}

File: src/BaseHmiTypes/Screens/Widgets/HmiAlarmIndicator.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Widgets;

public class HmiAlarmIndicator : HmiSimpleScreenItemBase
{
    public HmiAlarmIndicator()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiAlarmIndicator;
    }

    public HmiProperty<bool>? IsFlashingRequired { get; set; }

    public HmiProperty<HmiColor>? FlashingColor { get; set; }

    public HmiProperty<int>? FlashingRate { get; set; }

    public HmiProperty<int>? AlarmState { get; set; }

    public HmiProperty<int>? NoAlarmState { get; set; }

    public HmiProperty<int>? NumberOfAlarms { get; set; }

    public HmiProperty<IList<int>>? ShowAcknowledgedAlarmClasses { get; set; }

    public HmiProperty<IList<int>>? ShowPendingAlarmClasses { get; set; }
}
